### PR TITLE
feat: JDK 17 and Jakarta EE 9 support

### PR DESCRIPTION
**Pull request by MarcinVaadin**
## Description

Dependencies update after https://github.com/vaadin/quarkus/pull/83

## Type of change

- [ ] Bugfix
- [x] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [x] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

**Review by mshabarov — CHANGES_REQUESTED** [review]
This should be `V24` branch. Or is this only for testing?

**Comment by mshabarov** [comment]
Needs `com.vaadin:vaadin-quarkus:jar:2.0-SNAPSHOT` after https://github.com/vaadin/quarkus/pull/83 and snapshot release.

**Comment by MarcinVaadin** [comment]
> This should be `V24` branch. Or is this only for testing?

PR will be updated after quarkus 2.0-SNAPSHOT become available and merged into new v24 branch, I've created it too early.

File: pom.xml [hunk 1]
```diff
@@ -9,18 +9,19 @@
...
+        <maven.compiler.source>17</maven.compiler.source>
 
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
         <maven.compiler.parameters>true</maven.compiler.parameters>
         <failOnMissingWebXml>false</failOnMissingWebXml>
 
-        <vaadin.version>23.2.6</vaadin.version>
+        <vaadin.version>23.2-SNAPSHOT</vaadin.version>
```
Comment (mshabarov) [inline pom.xml]: Use platform 24.0-SNAPSHOT

**Review by mshabarov — CHANGES_REQUESTED** [review]

**Review by mshabarov — APPROVED** [review]
Looks good to me! Push it to `v24`

**Pull request closed without merging**